TSPL.docc/LanguageGuide/Concurrency.md

```diff
@@ -1591,6 +1591,370 @@ You can also use an unavailable conformance
+    }
+}
+
+Task { @MainActor in
+    let ballerina = Ballerina()
+    performIfDancer(ballerina)  // Prints "Ballerina.perform"
+}
+
+Task { @concurrent in
+    let ballerina = Ballerina()
+    performIfDancer(ballerina)  // Prints nothing
+}
+```
+
+In the above code,
+the call to `performIfDancer` from a main-actor isolated task
+matches the isolation of the conformance,
+so the dynamic cast succeeds.
+The call to `performIfDancer` from a concurrent task
+happens outside the main actor,
+so the dynamic cast fails and `perform` is not called.
+
+#### Restricting Isolated Conformances in Concurrent Code
+
+Protocol requirements can be used
+through instances of conforming types and through
+instances of the conforming types themselves,
+called *metatype values*.
+In generic code,
+a conformance requirement to `Sendable` or `SendableMetatype`
+tells Swift that an instance or metatype value is safe to use concurrently.
+To prevent isolated conformances from being used outside of their actor,
+a type with an isolated conformance
+can't be used as the concrete generic argument for a type
+parameter that requires a conformance
+to `Sendable` or `SendableMetatype`.
+
+A conformance requirement to `Sendable` indicates
+that instances may be passed across isolation boundaries and used concurrently:
+
+```swift
+protocol Dancer {
+    func perform()
+}
+
+func performConcurrently<D: Dancer>(_ dancer: D) where D: Sendable {
+    Task { @concurrent in
+        dancer.perform()
+    }
+}
+```
+
+The above code would admit data races
+if the conformance to `Dancer` was isolated,
+because the implementation of `perform`
+may access global actor isolated state.
+To prevent data races,
+Swift prohibits using an isolated conformance
+when the type is also required to conform to `Sendable`:
+
+```swift
+@MainActor class Ballerina: @MainActor Dancer { ... }
+
+let ballerina = Ballerina()
+performConcurrently(ballerina)  // Error
+```
+
+The above code results in an error
+because the conformance of `Ballerina` to `Dancer` is main-actor isolated,
+which can't satisfy the `Sendable` requirement of `performConcurrently`.
+
+Static and initializer protocol requirements
+can also be called through metatype values.
+A conformance to Sendable on the metatype type,
+such as `Int.Type`,
+indicates that a metatype value is safe
+to pass across isolation boundaries and used concurrently.
+Metatype types can conform to `Sendable`
+even when the type does not conform to `Sendable`;
+this means that only metatype values are safe to share in concurrent code,
+but instances of the type are not.
+
+In generic code,
+a conformance requirement to `SendableMetatype`
+indicates that the metatype of a type conforms to `Sendable`,
+which allows the implementation to share metatype values in concurrent code:
+
+```swift
+protocol Dancer {
+    init()
+    func perform()
+}
+
+func performConcurrently<D: Dancer>(n: Int, for: D.Type) async where T: SendableMetatype {
```

Comment:
I think the `T` is just a leftover from a previous version that had different type names?

```suggestion
func performConcurrently<D: Dancer>(n: Int, for: D.Type) async where D: SendableMetatype {
```